docs/sphinx/source/user_guide/weather_data.rst

```diff
@@ -0,0 +1,159 @@
+(e.g. precipitation data from :py:func:`~pvlib.iotools.get_acis_prism`
+for soiling modeling, and horizon profiles from :py:func:`~pvlib.iotools.get_pvgis_horizon`
+for horizon shade modeling).
+
+For a complete list of functions related to retrieving and importing weather
+data, see :ref:`iotools`.
+
+
+Types of weather data sources
+-----------------------------
+
+Weather data can be grouped into a few fundamental categories.  Which
+type is most useful depends on the application.  Here we provide a high-level
+overview of different types of weather data, and when you might want to use
+them.
+
+1. **Ground station measurements**:
+   From in-situ monitoring equipment. If properly maintained and
+   quality-controlled, these are the highest quality
+   source of weather information. However, the coverage depends on
+   a weather station having been set up in advance for the location and
+   times of interest. Some ground station networks like the BSRN and SURFRAD
+   make their measurement data publicly available.
+   
+   Data from public ground station measurement networks are useful if you
+   want accurate, high-resolution data but have flexibility around the
+   specific measurement location.
+
+2. **Satellite data**: 
+   These sources process satellite imagery (typically from geostationary
+   satellites) to identify and classify clouds, and combine this with solar
+   irradiance models to produce irradiance estimates. The quality is
+   generally much higher than NWP, but still not as good as a well-maintained
+   weather station. They have high spatial and temporal resolution
+   corresponding to the source satellite imagery, and are generally
+   optimised to estimate solar irradiance for PV applications. Free sources
+   such as PVGIS are available, and commerical sources such as SolarAnywhere,
+   Solcast and Solargis provide paid options though often have free trials.
+   
+   Satellite data is useful when suitable ground station measurements are
+   not available for the location and/or times of interest.
+
+3. **Numerical Weather Prediction (NWP)**:
+   These are mathematical simulations of weather systems.
+   The data quality is much lower than that of measurements and
+   satellite data, owing in part to coarser spatial and temporal
+   resolution, as well as many models not being optimised for solar
+   irradiance for PV applications. On the plus side, these models typically
+   have worldwide coverage, with some regional models (e.g. HRRR) sacrifice
+   global coverage for somewhat higher spatial and temporal resolution.
+   Various forecast (e.g. GFS, ECMWF, ICON) and reanalysis sources (ERA5,
+   MERRA2) exist.
+   
+   NWP datasets are primarily useful for parts of the world not covered
+   by satellite-based datasets (e.g. the poles) or if extremely long time
+   ranges are needed.
+
+For a more detailed comparison of the weather datasets available through
+pvlib, see [1]_.
```

Comment:
Can we move this citation up to the intro (I doubt most folks would read this far)